providers/fips/self_test_kats.c

```diff
@@ -149,6 +149,9 @@ static int self_test_kdf(const ST_KAT_KDF *t, OSSL_ST_EVENT *event,
 
     SELF_TEST_EVENT_onbegin(event, OSSL_SELF_TEST_TYPE_KAT_KDF, t->desc);
 
+    /* Terminate the array just in case we exit early */
+    params[0] = OSSL_PARAM_construct_end();
```

Comment:
This doesn't seem to quite work. If we exit early half way through the loop below, params will still not be terminated.

Reply:
Changed to clear the params (also added assert that we dont write to/past the end of the params array)